opencv-glib/image.cpp

```diff
@@ -201,6 +201,274 @@ gcv_drawing_options_new(void)
+  PROP_ITERATIONS,
+  PROP_KSIZE,
+  PROP_KTYPE,
+  PROP_MAX_LEVEL,
+  PROP_ANCHOR,
+};
+
+static void
+gcv_image_filtering_options_get_property(GObject *object,
+                                 guint prop_id,
+                                 GValue *value,
+                                 GParamSpec *pspec)
+{
+  auto priv = GCV_IMAGE_FILTERING_OPTIONS_GET_PRIVATE(object);
+
+  switch (prop_id) {
+  case PROP_NORMALIZE:
+    g_value_set_boolean(value, priv->normalize);
+    break;
+/* TODO
+  case PROP_BORDER_VALUE:
+    g_value_set_XXX(value, priv->border_value);
+    break;
+  case PROP_DSTSIZE:
+    g_value_set_XXX(value, priv->dstsize);
+    break;
+*/
+  case PROP_DELTA:
+    g_value_set_double(value, priv->delta);
+    break;
+  case PROP_PSI:
+    g_value_set_double(value, priv->psi);
+    break;
+  case PROP_SCALE:
+    g_value_set_double(value, priv->scale);
+    break;
+  case PROP_SIGMA_Y:
+    g_value_set_double(value, priv->sigma_y);
+    break;
+  case PROP_BORDER_TYPE:
+    g_value_set_enum(value, priv->border_type);
+    break;
+  case PROP_ITERATIONS:
+    g_value_set_int(value, priv->iterations);
+    break;
+  case PROP_KSIZE:
+    g_value_set_int(value, priv->ksize);
+    break;
+/*
+  case PROP_KTYPE:
+    g_value_set_int(value, priv->ktype);
+    break;
+*/
+  case PROP_MAX_LEVEL:
+    g_value_set_int(value, priv->max_level);
+    break;
+/*
+  case PROP_ANCHOR:
+    g_value_set_XXX(value, priv->anchor);
+    break;
+*/
+  default:
+    G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+    break;
+  }
+
+}
+
+static void
+gcv_image_filtering_options_set_property(GObject *object,
+                                 guint prop_id,
+                                 const GValue *value,
+                                 GParamSpec *pspec)
+{
+  auto priv = GCV_IMAGE_FILTERING_OPTIONS_GET_PRIVATE(object);
+
+  switch (prop_id) {
+  case PROP_NORMALIZE:
+    priv->normalize = g_value_get_boolean(value);
+    break;
+/* TODO
+  case PROP_BORDER_VALUE:
+    priv->border_value = g_value_get_XXX(value);
+    break;
+  case PROP_DSTSIZE:
+    priv->dstsize = g_value_get_XXX(value);
+    break;
+*/
+  case PROP_DELTA:
+    priv->delta = g_value_get_double(value);
+    break;
+  case PROP_PSI:
+    priv->psi = g_value_get_double(value);
+    break;
+  case PROP_SCALE:
+    priv->scale = g_value_get_double(value);
+    break;
+  case PROP_SIGMA_Y:
+    priv->sigma_y = g_value_get_double(value);
+    break;
+  case PROP_BORDER_TYPE:
+    priv->border_type = static_cast<GCVBorderType>(g_value_get_enum(value));
+    break;
+  case PROP_ITERATIONS:
+    priv->iterations = g_value_get_int(value);
+    break;
+  case PROP_KSIZE:
+    priv->ksize = g_value_get_int(value);
+    break;
+/*
+  case PROP_KTYPE:
+    priv->ktype = g_value_get_int(value);
+    break;
+*/
+  case PROP_MAX_LEVEL:
+    priv->max_level = g_value_get_int(value);
+    break;
+/*
+  case PROP_ANCHOR:
+    priv->anchor = g_value_get_XXX(value);
+    break;
```

Comment:
In this part, I would like to implement the following.
Could you advise for that?
(Thank you for helping me. I'm not familiar with C++ pointers, by the way.)

```ruby
options = CV::ImageFilteringOptions.new
options.anchor = CV::Point.new(-1,-1)
#...
options.anchor
```